### PR TITLE
Add CollectdError exception which can be thrown without causing a sta…

### PR DESCRIPTION
**Pull request by trenkel**
This should address #2330.

### CODE REVIEW

**Comment by rubenk** [comment]
Thanks Sven, this looks good. Could you add a short description about this to collectd-python(5) as well?

File: src/python.c [hunk 1]
```diff
@@ -318,11 +325,17 @@ void cpy_log_exception(const char *context) {
     typename = "NamelessException";
   if (message == NULL)
     message = "N/A";
-  Py_BEGIN_ALLOW_THREADS ERROR("Unhandled python exception in %s: %s: %s",
```
Comment (octo) [inline src/python.c]: This unfortunate formatting is very likely clang-format's fault. Can we add a semi-colon after the macro to put a stop to this?

Comment (trenkel) [inline src/python.c]: I'll add those if it happens again.

**Comment by trenkel** [comment]
Done.

File: src/collectd-python.pod [hunk 2]
```diff
@@ -248,6 +250,18 @@ collectd you're done.
...
+=head2 CollectdError
+
+This is an exception. If any Python script raises this exception it will
+still be treated like an error by collectd but it will be logged as a
+warning instead of an error and it will never generate a stacktrace.
+
+ class CollectdError(Exception)
+
+Basic exception for collectd Python scripts.
+Throwing this exception will not cause a stacktrace to be logged, even if
```
Comment (octo) [inline src/collectd-python.pod]: Is it possible to create a custom exception which is a subclass of this class?

File: src/python.c (hunk 1 above)
Comment (rpv-tomsk) [reply src/python.c]: I'll add these semi-colons...

**Comment by rpv-tomsk** [comment]
Merged.
Thanks for your module and its improvement, Sven!

**Pull request merged**